extensions/amp-wpm-player/0.1/amp-wpm-player.js

```diff
@@ -0,0 +1,557 @@
+          value => JSON.parse(decodeURIComponent(value))
+      );
+
+    /**
+     * Method that parses a boolean from the html attribute
+     * specified in the name parameter
+     * @private
+     * @param {string} name
+     * @param {boolean} required Specifies weather to throw and error when the attribute is not preset
+     *
+     * @return {boolean}
+     */
+    parseAttribute.boolean = (name, required) =>
+      parseAttribute(
+          name,
+          required,
+          value => value.toLowerCase() === 'true',
+      );
+
+    /**
+     * Method that parses a string from the html attribute
+     * specified in the name parameter
+     * @private
+     * @param {string} name
+     * @param {boolean} required Specifies weather to throw and error when the attribute is not preset
+     *
+     * @return {string}
+     */
+    parseAttribute.string = (name, required) =>
+      parseAttribute(
+          name,
+          required,
+          value => value,
+      );
+
+    /**
+     * Method that parses a number from the html attribute
+     * specified in the name parameter
+     * @private
+     * @param {string} name
+     * @param {boolean} required Specifies weather to throw and error when the attribute is not preset
+     *
+     * @return {number}
+     */
+    parseAttribute.number = (name, required) =>
+      parseAttribute(
+          name,
+          required,
+          value => parseInt(value, 10),
+      );
+
+    const clip = parseAttribute.json('clip') || {};
+    return {
+      ampcontrols: true,
+      forceUrl4stat: this.win.location.href,
+      target: 'playerTarget',
+      ...clip,
+      adv: parseAttribute.boolean('adv'),
+      url: parseAttribute.string('url'),
+      title: parseAttribute.string('title'),
+      screenshot: parseAttribute.string('screenshot'),
+      forcerelated: parseAttribute.boolean('forcerelated'),
+      hiderelated: parseAttribute.boolean('hiderelated'),
+      hideendscreen: parseAttribute.boolean('hideendscreen'),
+      mediaEmbed: parseAttribute.string('mediaEmbed'),
+      extendedrelated: parseAttribute.boolean('extendedrelated'),
+      skin: parseAttribute.json('skin'),
+      showlogo: parseAttribute.boolean('showlogo'),
+      watermark: parseAttribute.boolean('watermark'),
+      qoeEventsConfig: parseAttribute.json('qoeEventsConfig'),
+      advVastDuration: parseAttribute.number('advVastDuration'),
+      vastTag: parseAttribute.string('vastTag'),
+      embedTrackings: parseAttribute.json('embedTrackings'),
+      id: parseAttribute.string('id'),
+
+      autoplay: parseAttribute.boolean(VideoAttributes.AUTOPLAY) || false,
+      ampnoaudio: parseAttribute.boolean(VideoAttributes.NO_AUDIO),
+      dock: parseAttribute.boolean(VideoAttributes.DOCK),
+      rotateToFullscreen: parseAttribute.boolean(
+          VideoAttributes.ROTATE_TO_FULLSCREEN,
+      ),
+    };
+  }
+
+  /**
+   * Method that sends postMessage to iframe that contains the player.
```

Comment:
nit: "Method that..." is redundant as it is in every method, please just start with the verb.